similarity-data-rest/src/main/java/org/phenotips/data/similarity/rest/PatientMatchResource.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+package org.phenotips.data.similarity.rest;
+
+import org.phenotips.data.rest.PatientResource;
+import org.phenotips.rest.ParentResource;
+import org.phenotips.rest.Relation;
+
+import org.xwiki.stability.Unstable;
+
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+/**
+ * Resource for working with patient match data.
+ *
+ * @version $Id$
+ * @since 1.1
+ */
+@Unstable("New API introduced in 1.1")
+@Path("/patients/{patient-id}/similar-cases")
```

Comment:
For 1.2 only, this will have to be updated to `{entity-id}`.

Reply:
does it affect anything but naming? Any reason not to do it right away?

Reply:
This code itself will work, but what won't work is linking to/from the patient JSON. The autolinker creates links from the patient record to this "child" resource, and from this resource to the parent. Since the parent `/patient/` resource has a parameter, computing the URL requires knowing what is the value for that parameter, and since the parameters have different names, the autolinker won't know that the value for `entity-id` is supposed to be used for `patient-id`. The reverse is also true when linking from the patient to this resource.